src/Mod/Sketcher/App/Sketch.cpp

```diff
@@ -1727,6 +1727,25 @@ int Sketch::addConstraint(const Constraint *constraint)
             rtn = addTangentConstraint(constraint->First,constraint->Second);
         }
         else {
+            // check for B-Spline Knot to curve tangency
+            if(constraint->Third == GeoEnum::GeoUndef && constraint->SecondPos == PointPos::none && constraint->FirstPos == PointPos::start) {

```

Comment:
`Third` is what stores the knot point, unless I'm missing something. So it can't be `GeoUndef`